include/openrave/kinbody.h

```diff
@@ -3066,6 +3083,25 @@ class OPENRAVE_API KinBody : public InterfaceBase
     ConfigurationSpecification _spec;
     CollisionCheckerBasePtr _selfcollisionchecker; ///< optional checker to use for self-collisions
 
+public:
+    // Gary registers forward kinematics functions
+    struct ForwardKinematicsStruct {
+        ForwardKinematicsStruct();
+        ModuleBaseConstPtr pCalculatorModule = nullptr; ///< kinbody basic calculators module
+        boost::function<bool(const std::vector<double>&)> pSetLinkTransformsFn; ///< function that sets links' transforms
```

Comment:
@tgn3000 use `dReal` instead of `double`

Reply:
replaced